src/schemata/definitions/field.yml

```diff
@@ -67,6 +72,55 @@ allOf:
       - $ref: '#/definitions/field-schema/relationship'
       - $ref: '#/definitions/field-schema/script'
       - $ref: '#/definitions/field-schema/timer'
+  # only allow text & textarea widgets for string fields
+  - if:
```

Comment:
This is where we would add a `description` to express the constraint the schema implements. I would likely lean to only adding it to `if` and not split it up between `if` and `then`, since they both belong together anyway.

Reply:
This already looks much better :+1: 

Interesting though that you put the type inside `then` and the configuration inside `if`.. I would think that it make more sense to do it the other way around, but not sure if it works, e.g.

```yml
if:
  type: string
then:
  ui:widget:
    enum: ['text', 'textarea']
```

Reply:
Yes, we can flip this